---
title: "Count_Matrix_Stats.Filter"
author: "Samuel Gurr"
date: "February 1, 2021"
output: html_document
---

# Setup: 

```{r setup, include=TRUE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE, cache = TRUE)
```

### Load libraries
```{r  setup, include=FALSE}

library(dplyr)
library(edgeR)
library(cowplot)
library(ggplot2)
library(reshape2)
library(stringr)
library(tidyr)
library(edgeR)

# SET WORKING DIRECTORY

knitr::opts_knit$set(root.dir = "C:/Users/samjg/Documents/Github_repositories/Cvirginica_KM_paper/RAnalysis") # sets the working 

```


# LOAD DATA:

```{r  load_data}

raw_counts_mtx  <- read.table(file="Data/gene_count_star.txt", sep='\t', header=TRUE) 
Metadata        <- read.csv(file="Data/SummaryforWGCNAtreat.csv", sep=',', header=TRUE)

```


# Format to sampling days

```{r  format data}

# get the IDs to truncate columns of the raw gene count matrix
Metadata_D0 <- Metadata %>%  dplyr::filter(Day == 0) # metadata for time 0
Metadata_D1 <- Metadata %>%  dplyr::filter(Day == 1) # metadata for time 1
# abundance matrix
raw_counts_mtx # this is all the data
raw_counts_mtx_D0 <- raw_counts_mtx %>% 
  dplyr::select(c('gene_id', Metadata_D0$Sample.ID)) # select sample IDs for day 0
raw_counts_mtx_D1 <- raw_counts_mtx %>% 
  dplyr::select(c('gene_id', Metadata_D1$Sample.ID)) # select sample IDs for day 1

```


Filter read counts 
About: usiing edgeR function 'cpm' here to filter by counts bper million 

Below I am making two cut-offs (thresholds) as (1) lenient low-end (2) strict/more conservative 
(1) < 3 counts per million in at least 50% of samples 
(2) < 10 counts per million in at least 50% of samples 

Resulting umber of genes summarized in knitr table and output to TagSeq\Analysis\Data\filtered_counts'

For loop in 1 CPM increments to optimize the filtering threshold 
Output is a grid filture for  days 2 and 18


```{r CPM LOOP - Rarefaction-like approach to read filtering threshold}

# For loop ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::;;; #
df_total   <- data.frame() # start dataframe 
loop.table <- data.frame(matrix(nrow = 3, ncol = 4)) # create dataframe, columns for the number of sampling days
colnames(loop.table)<-c('CPM_threshold', 'Day', 'total_genes', 'mean_reads_per_gene') # names for comuns in the for loop
CPM_loop <- c(0:50) # run the loop for CPM from 1 - 10 (counts per million) 
CPM_loop <- data.frame(CPM_loop)

for (i in 1:nrow(CPM_loop)) {
  
  var <- CPM_loop[i,1]
  
  # ========================================================== 
  #
  # All
  # ========================================================== 
  cts.All        <- data.frame(raw_counts_mtx[,-1], row.names=raw_counts_mtx[,1])
  cts.All.matrix <- as.matrix(cts.All, row.names="transcript_id")
  CPM.All        <- cpm(cts.All.matrix) # Obtain CPMs (counts oer million) using egdeR
  thresh.All     <- CPM.All > var # Which values in myCPM are greater than 3?
  keep.All       <- rowSums(thresh.All) >= (ncol(thresh.All)/2) # we would like to keep genes that have at least 50% TRUES in each row of thresh
  rowSums(head(thresh.All)) # Summary of how many TRUEs there are in each row
  table(rowSums(thresh.All)) # 6880 genes with TRUE in all 36 samples 
  cts.matrix.All.LOOPED     <- cts.All.matrix[keep.All,]
  
  All_total_genes           <- nrow(cts.matrix.All.LOOPED)
  All_mean_reads_per_gene   <- mean( (rowSums(cts.matrix.All.LOOPED) / ncol(cts.matrix.All.LOOPED) ) ) 
  
  # ========================================================== 
  #
  # DAY 0
  # ========================================================== 
  cts.d0        <- data.frame(raw_counts_mtx_D0[,-1], row.names=raw_counts_mtx_D0[,1])
  cts.d0.matrix <- as.matrix(cts.d0, row.names="transcript_id")
  CPM.d0        <- cpm(cts.d0.matrix) # Obtain CPMs (counts oer million) using egdeR
  thresh.d0     <- CPM.d0 > var # Which values in myCPM are greater than 3?
  keep.d0       <- rowSums(thresh.d0) >= (ncol(thresh.d0)/2) # we would like to keep genes that have at least 50% TRUES in each row of thresh
  rowSums(head(thresh.d0)) # Summary of how many TRUEs there are in each row
  table(rowSums(thresh.d0)) # 6880 genes with TRUE in all 36 samples 
  cts.matrix.d0.LOOPED     <- cts.d0.matrix[keep.d0,]
  
  d0_total_genes           <- nrow(cts.matrix.d0.LOOPED)
  d0_mean_reads_per_gene   <- mean( (rowSums(cts.matrix.d0.LOOPED) / ncol(cts.matrix.d0.LOOPED) ) ) 
  
  # ========================================================== 
  #
  # DAY 1
  # ========================================================== 
  cts.d1        <- data.frame(raw_counts_mtx_D1[,-1], row.names=raw_counts_mtx_D1[,1])
  cts.d1.matrix <- as.matrix(cts.d1, row.names="transcript_id")
  CPM.d1        <- cpm(cts.d1.matrix) # Obtain CPMs (counts oer million) using egdeR
  thresh.d1     <- CPM.d1 > var # Which values in myCPM are greater than 3?
  keep.d1       <- rowSums(thresh.d1) >= (ncol(thresh.d1)/2) # we would like to keep genes that have at least 50% TRUES in each row of thresh
  rowSums(head(thresh.d1)) # Summary of how many TRUEs there are in each row
  table(rowSums(thresh.d1)) # 6880 genes with TRUE in all 36 samples 
  cts.matrix.d1.LOOPED     <- cts.d1.matrix[keep.d1,]
  
  d1_total_genes           <- nrow(cts.matrix.d1.LOOPED)
  d1_mean_reads_per_gene   <- mean( (rowSums(cts.matrix.d1.LOOPED) / ncol(cts.matrix.d1.LOOPED) ) ) 
  
    # table  
  loop.table$CPM_threshold           <- c(var, var, var)
  loop.table$Day                     <- c('all', 0, 1)
  loop.table$total_genes             <- c(All_total_genes, 
                                          d0_total_genes, 
                                          d1_total_genes)
  loop.table$mean_reads_per_gene   <- c(All_mean_reads_per_gene, 
                                        d0_mean_reads_per_gene, 
                                        d1_mean_reads_per_gene)
    
  df <- data.frame(loop.table) # name dataframe for this singl e row
  df_total <- rbind(df_total,df) #bind to a cumulative list dataframe
  #print(df_total) # print to monitor progress
}

df_total
```
# LETS SEE THE DATA SHALL WE!
```{r figs}

All_filterfig <- df_total %>% 
  dplyr::filter(Day %in% 'all') %>% 
  tidyr::gather(key = "variable", value = "measurement",
       total_genes, mean_reads_per_gene, -Day) %>% 
  # Start with a usual ggplot2 call:
  ggplot(aes(x = CPM_threshold)) +
  geom_line(aes(y = measurement)) +
  ggtitle("All data") +
  geom_vline(xintercept = c(3,5,10), color = 'red') +
  facet_wrap(. ~ variable, scales="free")
# All_filterfig

D0_filterfig <- df_total %>% 
  dplyr::filter(Day %in% 0) %>% 
  tidyr::gather(key = "variable", value = "measurement",
       total_genes, mean_reads_per_gene, -Day) %>% 
  # Start with a usual ggplot2 call:
  ggplot(aes(x = CPM_threshold)) +
  geom_line(aes(y = measurement)) +
  ggtitle("Day 0") +
  geom_vline(xintercept = c(3,5,10), color = 'red') +
  facet_wrap(. ~ variable, scales="free")
# D0_filterfig

D1_filterfig <- df_total %>% 
  dplyr::filter(Day %in% 1) %>% 
  tidyr::gather(key = "variable", value = "measurement",
       total_genes, mean_reads_per_gene, -Day) %>% 
  # Start with a usual ggplot18 call:
  ggplot(aes(x = CPM_threshold)) +
  geom_line(aes(y = measurement)) +
  ggtitle("Day 1") +
  geom_vline(xintercept = c(3,5,10), color = 'red') +
  facet_wrap(. ~ variable, scales="free")
# D1_filterfig



png("C:/Users/samjg/Documents/Github_repositories/Cvirginica_KM_paper/RAnalysis/Output/CPM_Read_Filter_Curve.png", 2000, 1000, pointsize=20)
plot_grid(All_filterfig, D0_filterfig, D1_filterfig)
dev.off()

```


### Results of Rarefaction-like nonbias approach to read filtering
Rationale: in the previous cluster, CPMs 1 - 50 were run (including 'in 50% of samples' due to the binary wild v. aquaculture present in half of ALL matrices)
in a for loop to produce a rare-faction-like curve for read filtering. Observe the output "../Data/Filtered_Counts/CPM_Read_Filter_Curve.png"
Figure shows that the 5CPM threshold looks like the best filter for consistant criteria in the folloing:
a) close to a signiicant change in the regression when low reads were filtered out
b) all genes have >250 mean read count sample-1
c) 4.6-4.8k genes selected across matrices - ~50-60% decrease in the total unigene count with single read (~10k)

LENIENT CUT-OFF [ 5 CPM IN 50% SAMPLES ]
```{r  5 CPM 50% SAMPLES}

# ========================================================== 
#
# DAY 0  (5 CPM in 50% samples using edgeR)
# ========================================================== 
raw.cts.all          <- data.frame(raw_counts_mtx[,-1], row.names=raw_counts_mtx[,1])
raw.countMatrix.all  <- as.matrix(raw.cts.all, row.names="transcript_id")
ncol(raw.countMatrix.all) # 24  samples from just Day 0
nrow(raw.countMatrix.all) # 38977 total genes
# pre-filtering; genes omitted if < 5 counts per million reads in 50% of samples
# view the colSums of our Day0 samples  - notice the read sums are around 1 million
CPM.all <- cpm(raw.countMatrix.all) # Obtain CPMs (counts oer million) using egdeR
head(CPM.all) # Have a look at the output
thresh.all <- CPM.all > 5 # Which values in myCPM are greater than 5?
head(thresh.all) # This produces a logical matrix with TRUEs and FALSES
rowSums(head(thresh.all)) # Summary of how many TRUEs there are in each row
table(rowSums(thresh.all)) # 8142 genes with TRUE in 12 samples; 21308 genes with false in 12 samples
#     0     1     2     3     4     5     6     7     8     9    10    11    12    13    14    15    16 
# 21308  1152   683   551   451   368   366   333   282   319   242   283   303   277   235   258   271 
#    17    18    19    20    21    22    23    24 
  # 324   376   337   351   429   561   775  8142 
keep.all <- rowSums(thresh.all) >= floor(ncol(thresh.all)/2) # we would like to keep genes that have at least 50% TRUES in each row of thresh
summary(keep.all) # only 8% of genes passed
#    Mode   FALSE    TRUE 
# logical   25881   13096
countMatrix.all.filtered <- raw.countMatrix.all[keep.all,] # Subset the rows (genes)
dim(countMatrix.all.filtered) # 12639  genes 


# ========================================================== 
#
# DAY 0  (5 CPM in 50% samples using edgeR)
# ========================================================== 
raw.cts.d0          <- data.frame(raw_counts_mtx_D0[,-1], row.names=raw_counts_mtx_D0[,1])
raw.countMatrix.d0  <- as.matrix(raw.cts.d0, row.names="transcript_id")
ncol(raw.countMatrix.d0) # 12  samples from just Day 0
nrow(raw.countMatrix.d0) # 38977 total genes
# pre-filtering; genes omitted if < 5 counts per million reads in 50% of samples
# view the colSums of our Day0 samples  - notice the read sums are around 1 million
CPM.d0 <- cpm(raw.countMatrix.d0) # Obtain CPMs (counts oer million) using egdeR
head(CPM.d0) # Have a look at the output
thresh.d0 <- CPM.d0 > 5 # Which values in myCPM are greater than 5?
head(thresh.d0) # This produces a logical matrix with TRUEs and FALSES
rowSums(head(thresh.d0)) # Summary of how many TRUEs there are in each row
table(rowSums(thresh.d0)) # 9017 genes with TRUE in 12 samples; 21991 genes with false in 12 samples
#     0     1     2     3     4     5     6     7     8     9    10    11    12 
# 21991  1239   818   696   580   557   742   550   524   639   682   942  9017 
keep.d0 <- rowSums(thresh.d0) >= floor(ncol(thresh.d0)/2) # we would like to keep genes that have at least 50% TRUES in each row of thresh
summary(keep.d0) # only 8% of genes passed
#    Mode   FALSE    TRUE 
# logical   25881   13096
countMatrix.d0.filtered <- raw.countMatrix.d0[keep.d0,] # Subset the rows (genes)
dim(countMatrix.d0.filtered) # 13096  genes 



# ========================================================== 
#
# DAY 1  (5 CPM in 51% samples using edgeR)
# ========================================================== 
raw.cts.d1          <- data.frame(raw_counts_mtx_D1[,-1], row.names=raw_counts_mtx_D1[,1])
raw.countMatrix.d1  <- as.matrix(raw.cts.d1, row.names="transcript_id")
ncol(raw.countMatrix.d1) # 12  samples from just Day 1
nrow(raw.countMatrix.d1) # 38977 total genes
# pre-filtering; genes omitted if < 5 counts per million reads in 51% of samples
# view the colSums of our Day1 samples  - notice the read sums are around 1 million
CPM.d1 <- cpm(raw.countMatrix.d1) # Obtain CPMs (counts oer million) using egdeR
head(CPM.d1) # Have a look at the output
thresh.d1 <- CPM.d1 > 5 # Which values in myCPM are greater than 5?
head(thresh.d1) # This produces a logical matrix with TRUEs and FALSES
rowSums(head(thresh.d1)) # Summary of how many TRUEs there are in each row
table(rowSums(thresh.d1)) # 9191 genes with TRUE in 12 samples; 23016 genes with false in 12 samples
#     0     1     2     3     4     5     6     7     8     9    10    11    12 
# 23016  1210   732   548   468   447   566   453   411   470   598   867  9191
keep.d1 <- rowSums(thresh.d1) >= floor(ncol(thresh.d1)/2) # we would like to keep genes that have at least 51% TRUES in each row of thresh
summary(keep.d1) # only 8% of genes passed
#    Mode   FALSE    TRUE 
# logical   25881   13196
countMatrix.d1.filtered <- raw.countMatrix.d1[keep.d1,] # Subset the rows (genes)
dim(countMatrix.d1.filtered) # 12556  genes 


# write csv
write.csv(countMatrix.all.filtered,"C:/Users/samjg/Documents/Github_repositories/Cvirginica_KM_paper/RAnalysis/Output/Filtering_reads/All.filtered_5cpm50perc.csv") # 'path' called in previous # write .csv section

write.csv(countMatrix.d0.filtered,"C:/Users/samjg/Documents/Github_repositories/Cvirginica_KM_paper/RAnalysis/Output/Filtering_reads/D0.filtered_5cpm50perc.csv") # 'path' called in previous # write .csv section

write.csv(countMatrix.d1.filtered,"C:/Users/samjg/Documents/Github_repositories/Cvirginica_KM_paper/RAnalysis/Output/Filtering_reads/D1.filtered_5cpm50perc.csv") # 'path' called in previous # write .csv section


```


# Post-filtered (<5CPM in 50% samples) assessment of 'cleaned' read matrix (lolwy-expressed removed!)

* counts per sample 
* unigenes represented (total and % to reference)

```{r stats on RAW count matrix}

d2_filtered_counts          <- read.csv(file="C:/Users/samjg/Documents/Github_repositories/Cvirginica_multistressor/RAnalysis/Data/TagSeq/Filtered_Counts/filtered_counts_5cpm_50perc/day2.filtered_5cpm50perc.csv", sep=',', header=TRUE)

d18_filtered_counts         <- read.csv(file="C:/Users/samjg/Documents/Github_repositories/Cvirginica_multistressor/RAnalysis/Data/TagSeq/Filtered_Counts/filtered_counts_5cpm_50perc/day18.filtered_5cpm50perc.csv", sep=',', header=TRUE)

d2_filtered_counts.hightemp <- read.csv(file="C:/Users/samjg/Documents/Github_repositories/Cvirginica_multistressor/RAnalysis/Data/TagSeq/Filtered_Counts/filtered_counts_5cpm_50perc/day2.filtered_5cpm50perc_hightemp.csv", sep=',', header=TRUE)


# day 2 ================================= #
# how many reads total?
sum(d2_filtered_counts[,-1]) # 33997765
# how many read counts per sample? (average and sd)
min(colSums(d2_filtered_counts[,-1])) # 789376
max(colSums(d2_filtered_counts[,-1])) # 2113582
mean(colSums(d2_filtered_counts[,-1])) # 1416574
sd(colSums(d2_filtered_counts[,-1])) # 337247
# how many unigenes with at a least a single read?
d2_sums_gtr0 <- rowSums(d2_filtered_counts[,-1]) > 0 # boleean of TRUE and FALSE for rows (transcript IDs) with > 0 reads
sum(d2_sums_gtr0 == TRUE) # 4820 total genes with unique transcript reads 
( sum(d2_sums_gtr0 == TRUE) / (dim(raw.countDataframe)[1])) *100 # 7.295182 % of genes have a unique mapped read - relative to the complete genome!


# day 18 ================================= #
# how many reads total?
sum(d18_filtered_counts[,-1]) # 16533993
# how many read counts per sample? (average and sd)
min(colSums(d18_filtered_counts[,-1])) # 711736
max(colSums(d18_filtered_counts[,-1])) # 2138741
mean(colSums(d18_filtered_counts[,-1])) # 1503090
sd(colSums(d18_filtered_counts[,-1])) # 360755.6
# how many unigenes with at a least a single read?
d18_sums_gtr0 <- rowSums(d18_filtered_counts[,-1]) > 0 # boleean of TRUE and FALSE for rows (transcript IDs) with > 0 reads
sum(d18_sums_gtr0 == TRUE) # 4936 total genes with unique transcript reads 
( sum(d18_sums_gtr0 == TRUE) / (dim(raw.countDataframe)[1])) *100 # 7.470751 % of genes have a unique mapped read - relative to the complete genome!


# mean read counts for all filtered samples day 2 and day 18 larvae and spat!
mean(c(colSums(d2_filtered_counts[,-1]), colSums(d18_filtered_counts[,-1]))) # 1443765
sd(c(colSums(d2_filtered_counts[,-1]), colSums(d18_filtered_counts[,-1]))) # 341873
```



